### PR TITLE
Remove use of kokkos arith traits in batched linear algebra

### PR DESCRIPTION
**Pull request by sslattery**
This largely fixes some issues that came out of using the Sacado FAD types and type conversions.

### CODE REVIEW

File: src/Picasso_BatchedLinearAlgebra.hpp [hunk 1]
```diff
@@ -1525,8 +1521,7 @@ KOKKOS_INLINE_FUNCTION auto diagonal( const ExpressionX& x )
                                   ExpressionX::extent_0, ExpressionX::extent_0>(
         [=]( const int i, const int j ) {
             return ( i == j ) ? x( i )
-                              : Kokkos::ArithTraits<
-                                    typename ExpressionX::value_type>::zero();
+                              : 0.0;
```
Comment (dalg24) [inline src/Picasso_BatchedLinearAlgebra.hpp:1524]: I doubt that would work for non floating point

Comment (sslattery) [reply src/Picasso_BatchedLinearAlgebra.hpp]: I don't expect to do integer arithmetic with this class

File: src/Picasso_BatchedLinearAlgebra.hpp [hunk 2]
```diff
@@ -1184,14 +1182,14 @@ operator*( const ExpressionA& a, const ExpressionB& b )
     typename ExpressionA::eval_type a_eval = a;
     typename ExpressionB::eval_type b_eval = b;
     Matrix<typename ExpressionA::value_type, ExpressionA::extent_0,
-           ExpressionB::extent_1>
-        c = Kokkos::ArithTraits<typename ExpressionA::value_type>::zero();
+           ExpressionB::extent_1> c =
+        static_cast<typename ExpressionA::value_type>(0.0);
```
[the remark this answers is not in the document]
Comment (dalg24) [reply src/Picasso_BatchedLinearAlgebra.hpp]: Constructor is explicit?

**Comment by sslattery** [comment]
Note we might not merge this as we are seeing downstream issues with Sacado types. I think we need to push FAD into Picasso anyway for various reasons so we would have better testing for this.

**Pull request closed without merging**